### PR TITLE
ECCI-275: Allows derivative images to be created.

### PR DESCRIPTION
**Pull request by siliconmeadow**

### CODE REVIEW

File: nginx-conf/nginx.conf [hunk 1]
```diff
@@ -131,11 +131,6 @@ http {
         }
 
         location ~* \.(js|css|png|jpg|jpeg|gif|ico|svg)$ {
-            # If we're outside of govuk then request no indexing, on the assumption that this is a non-production environment
```
Comment (Polynya) [inline nginx-conf/nginx.conf:134]: Should this be in its own commit?

File: patches/filesystem-chmod-0001.patch [hunk 2]
```diff
@@ -0,0 +1,15 @@
+--- a/core/lib/Drupal/Core/File/FileSystem.php	2023-02-27 12:38:42.368159781 +0000
++++ b/core/lib/Drupal/Core/File/FileSystem.php	2023-02-27 12:38:20.107897836 +0000
+@@ -106,6 +106,12 @@
+       return TRUE;
+     }
+ 
++    // We may get forced ogu+rwx permissions from Azure Blob Storage, consider this a success.
```
Comment (Polynya) [inline patches/filesystem-chmod-0001.patch:7]: Looks ok but is this the only place to fix the issue? Would be better if it could be fixed somewhere that only affects images

Comment (MatthewWilkes) [reply patches/filesystem-chmod-0001.patch:7]: There's are methods use this, which check if the chmod was a success. We could have changed that to ignore the failure, but decided to go for this one to help prevent other errors we haven't found yet. This method will always return failure on Azure blob storage, so seemed fair enough.

The thing I was wondering is if it would have been better to check if the mode was correct, and alter the config for target mode.

File: nginx-conf/nginx.conf (hunk 1 above)
Comment (MatthewWilkes) [reply nginx-conf/nginx.conf:134]: This is part of the fix, this if prevents the try_files working for images that don't exist in the FS so prevents the php controller kicking in. We did comment it out in the gov one instead.

**Review by MatthewWilkes — APPROVED** [review]

**Review by Polynya — APPROVED** [review]
